internal/scanner/scanner.go

```diff
@@ -89,27 +89,24 @@ func (s *Scanner) ScanNamespace(nsName string) error {
 	// create PolicyReport
 	namespacedsReport := report.NewPolicyReport(namespace)
 	namespacedsReport.Summary.Skip = skippedNum
+	// old policy report to be used as cache
+	previousNamespacedReport, err := s.reportStore.GetPolicyReport(nsName)
+	if err != nil {
+		log.Error().Err(err).Str("namespace", nsName).
+			Msg("error getting PolicyReport from store")
+	}
 
 	// Iterate through all auditableResources. Each item contains a list of resources and the policies that would need
 	// to evaluate them.
 	for i := range auditableResources {
-		auditResource(&auditableResources[i], &s.resourcesFetcher, &s.httpClient, &namespacedsReport)
-		err = s.reportStore.AddPolicyReport(&namespacedsReport)
+		auditResource(&auditableResources[i], s.resourcesFetcher, &s.httpClient, &namespacedsReport, &previousNamespacedReport)
```

Comment:
probably not for this PR, but I would like to refactor the `auditResource` function. I think it should become a method of `AuditableResource`, something like `evaluate` or `process`

Reply:
I agree. I've also suggested to do a change in the future adding an abstraction layer to avoid this duplicate function to handle Cluster and not cluster reports. They are almost the same.